aws/components/apigateway/setup.ftl

```diff
@@ -303,27 +303,64 @@
             outputs={}
             dependencies=apiId
         /]
+
+        [#-- Throttling Configuration --]
+        [#local methodSettings = []]
+        [#local openapiIntegrations = 
+            getOccurrenceSettingValue(occurrence, [["apigw"], ["Integrations"]], true)]
+
+        [#if openapiIntegrations.Throttling?has_content]
+            [#local methodSettings = [
+                {
+                    "HttpMethod": "*",
+                    "ResourcePath": "/*",
+                    "LoggingLevel": "INFO",
+                    "DataTraceEnabled": true
+                } +
+                attributeIfContent(
+                    "ThrottlingBurstLimit", 
+                    openapiIntegrations.Throttling.BurstLimit) +
+                attributeIfContent(
+                    "ThrottlingRateLimit",
+                    openapiIntegrations.Throttling.RateLimit)
+            ]]
+        [/#if]
+
+        [#-- Integration Patterns (as Regex) into Matching Method Throttling (as explicit paths) --]
+        [#if definitionsObject[core.Id]?? ]
+            [#list definitionsObject[core.Id].paths as path,pathConfig]
+                [#list pathConfig?keys as verb]
+                    [#list openapiIntegrations.Patterns as pattern]
+                        [#if path?matches(pattern.Path)] 
+                            [#local methodSettings += 
```

Comment:
Should only generate entry if the throttling element is defined in the pattern.

Reply:
Updated: 464cde1